tests/scripts/anchor-mardown-headings.test.js

```diff
@@ -76,6 +76,21 @@ test('anchorMarkdownHeadings', (t) => {
     t.equal(output, expected)
   })
 
+  t.test('correctly parses markdown heading with empty spaces', (t) => {
+    const text = '这是<a href="b">链接</a>的<a href="d">测试！</a>'
+    const level = 2
+    const raw = '<!-- empty  spaces - id -->这是[链接](b)c[测试！](d)'
```

Comment:
This comment will not match `ANCHOR_COMMENTREG`. Is it supposed to?

Reply:
Oh, wait, never mind, you updated `ANCHOR_COMMENTREG` to account for spaces.

Reply:
@Trott, I'm doing an experiment on win 10, x86. You can have a try by moving the `ANCHOR_COMMENTREG` out of the function as a global one, you will find it very strange not to pass the newly-added unit test (I'm NOT sure why it will add '-c' at the end of 'empty-spaces-id', but it won't if I use it directly in the function).

Reply:
@Trott, I found the problem, it's that if I use 'g', it will continue searching according to the lastIndex. But what makes me feel strange is that if so, only one unit test should pass, while the others fail. This problem is still in pending and disscussion, but this problem is fixed.